### ISSUE TITLE
Focus ring does not get removed after pressing enter on Buttons with tab navigation on Safari

### ISSUE THREAD

**Issue by raashidkhan1** [adobe/react-spectrum#2143]
# 🐛 Bug Report 

For keyboard accessibility, using tab navigation on Safari, go to Buttons and press enter, the focus ring on the button stays. Works fine in chrome, even the focus element is the button, focus ring should get removed on keydown/keyup.
## 🤔 Expected Behavior

Focus ring on the button should get removed on keydown/keyup with space/enter key.

## 😯 Current Behavior

Focus ring stays on the button.

## 💻 Code Sample
any react spectrum CTA button

https://codesandbox.io/s/sad-ride-ivn4y?file=/src/App.js

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 2.34
| Browser          | Safari
| Operating System | MacOS

## 🧢 Your Company/Team
Adobe/ CCX-Commenting

Adobe Photoshop Commenting Plugin on Webview.

**Comment by snowystinger** [comment]
This is a dupe of https://github.com/adobe/react-spectrum/issues/1935 and the codesandbox in there has the best way to handle it.
TL;DR you've disabled the button under the user, focus is returned to the body, it should be moved somewhere meaningful before that happens

on a side note, it's better to use props and state than directly setting attributes on the nodes themselves and searching for them each time

**Comment by snowystinger** [comment]
closing, reopen if it's different

**Comment by raashidkhan1** [comment]
> closing, reopen if it's still needed

@snowystinger Thanks for the info. I made the sandbox hurriedly just to mimic the behaviour of Safari. Would this issue be fixed or do I need to go with the workaround?

**Comment by snowystinger** [comment]
As stated in the issue that this is a duplicate of, here: https://github.com/adobe/react-spectrum/issues/1935#issuecomment-882755894
It's not really a concern because it's highlighting improper handling of focus. What I wrote in the codesandbox in that issue isn't a workaround, it's how focus should be handled.
